dulwich/objects.py

```diff
@@ -1084,6 +1072,30 @@ def format_timezone(offset, unnecessary_negative_timezone=False):
             (sign, offset / 3600, (offset / 60) % 60)).encode('ascii')
 
 
+def parse_time_entry(value):
+    """Parse time entry behavior
+
+    :param value: Bytes representing a git commit/tag line
+    :raise: ObjectFormatException in case of parsing error.
+    :return: Tuple of (author, time, (timezone, timezone_neg_utc))
+    """
+    try:
+        sep = value.index(b'> ')
+    except ValueError:
+        return (value, None, (None, False))
+    try:
+        person = value[0:sep+1]
+        rest = value[sep+2:]
+        timetext, timezonetext = rest.rsplit(b' ', 1)
+        time = int(timetext)
+        timezone, timezone_neg_utc = parse_timezone(timezonetext)
+        # checking for overflow error
+        datetime.datetime.fromtimestamp(time)
```

Comment:
This is an expensive way of checking the time, and depends on the implementation of datetime. Could you just check if the time was higher than the supported CGit value in .check() ?

Reply:
Sure, will look into that.
Thanks for the heads up!

Reply:
I need some help here, please. Here are my findings.

According to git fsck's code, [here is
the check on parsing the date and failing if error](https://github.com/git/git/blob/master/fsck.c#L695-L696)..

The definition of the function is in [date.c](https://github.com/git/git/blob/master/date.c#L1206-L1221).
This checks for a TIME_MAX value.

[Through the hide-and-seek game of header inclusion, we can then find the definition of that value](https://github.com/git/git/blob/master/git-compat-util.h#L328).
(from: date.c -> cache.h -> git-compat-util.h)

Its value is `UINTMAX_MAX` (unsigned integer max value).
So, is 2^32 sounds good enough?

Note: I interpreted the CGIT value you mentioned as the main git repository.
I'm not so sure i interpreted right though :)